assets/javascripts/discourse/controllers/upgrade.js

```diff
@@ -0,0 +1,27 @@
+import Controller from "@ember/controller";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+
+export default class Upgrade extends Controller {
+  @tracked banner = [];
```

Comment:
Could we use `TrackedArray` here to avoid needing to re-assign the `banner` property every time we change the array contents?